1152 custom fields tanslatable attributes

entantoencuanto opened the pull request:
Closes https://github.com/PopulateTools/issues/issues/1152


## :v: What does this PR do?

* Avoids errors in API and admin when custom field of type text/markdown changes from not translatable to translatable and vice versa.
* The API always returns a string and detects for not translatable fields if the payload is a hash and selects a value according to locale
* In the admin interface, in the edit form the values are taken from the payload (if not saved nothing changes in the record):
  * For fields changed from not translatable to translatable the content is stored in current locale and the others are kept blank
  * For fields changed from translatable to not translatable the value for the current locale is used in the text field.
These changes 

## :mag: How should this be manually tested?

Create translatable and not translatable text and long text custom fields. Fill the values on some resource using them. Edit the custom fields and flip the translatable feature.

## :shipit: Does this PR changes any configuration file?

No

(Changes in these files might need to update the role in Ansible)

## :book: Does this PR require updating the documentation?

No

entantoencuanto committed:
Avoid errors caused by changes in type in custom fields API cache

entantoencuanto committed:
Prevent errors in custom fields records raw values when translatable is changed

entantoencuanto committed:
Test API response when custom field type changes translatable ability

ferblape commented:
This is already released to staging but there are some tests failing

ferblape reviewed (commented):
Code is OK, just review the tests please

entantoencuanto committed:
Fix integration test

With the changes of this PR some new localized fields appear in the
admin form which produces an ambiguity error trying to switch locale.

entantoencuanto committed:
Fix CustomField model tests

entantoencuanto committed:
Merge branch 'master' into 1152-custom_fields_tanslatable_attributes

* master:
  Update rubocop to version 1.6.1
  Fix velocity missing require
  Bump ini from 1.3.5 to 1.3.7
  Only redirect the specific paths, avoiding the redirection with urls like /api/v1/dashboards
  fix locales
  name async
  set translations

entantoencuanto committed:
Fix missing tests
